src/components/SelectionList/Search/ActionCell.tsx

```diff
@@ -46,10 +61,30 @@ function ActionCell({onButtonPress, action = CONST.SEARCH.ACTION_TYPES.VIEW, isL
         );
     }
 
+    if (action === CONST.SEARCH.ACTION_TYPES.VIEW || action === CONST.SEARCH.ACTION_TYPES.REVIEW) {
+        return (
+            <Button
+                text={text}
+                onPress={goToItem}
+                small
+                pressOnEnter
+                style={[styles.w100]}
+            />
+        );
+    }
+
+    const command = SearchUtils.getTransactionActionCommand(action);
```

Comment:
This code actually got changed a little, so you might want to re-check it.
However the general idea that `goto` comes from a different source while commands/actions come from a different source is still here.

My one concern is performance. Unfortunately the component `SelectionList` is quite rigid and does not allow for any additional props or callbacks being passed when rendering items. Because of that **any extra data** we want to have available has to go inside the `items` themselves, which in our case means they have to go here: https://github.com/Expensify/App/blob/main/src/libs/SearchUtils.ts#L122 - which is quite heavy already. Every time the list renders and something changes in data, this whole function has to run and attach all the props.

We're passing down `goToItem` from Search because only search has full navigation context and this is also run on row click. 
But the actions don't need to be passed down from Search and are run only on this button - so I thought that solution is better. I'm really not that happy to make `TransactionListItemType` be aware of every single piece of data, including specific callbacks to run.

As for your point about reusability I agree in general, but in here Im not sure if we really need this component to be reusable? This is a Search specific component. It's `Button` and `Badge` that are nice and reusable. This component just composes them, and adds extra Search item logic.

What do you think?

Reply:
I think you make a good point. I'm also not a fan of adding more data to `getTransactionsSections`, but I agree that for the time being this is the solution we have.